### PR TITLE
Add FXIOS-7054 [v118] Show empty bottom sheet

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7054)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15681)

## :bulb: Description

- Present FakespotBottomSheetChildViewController using BottomSheetViewController
- Record telemetry events: 
          - shoppingAddressBarIconClicked
          - shoppingSurfaceClosed
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift [hunk 1]
```diff
@@ -109,6 +110,15 @@ extension BrowserViewController: URLBarDelegate {
         }
     }
 
+    func urlBarDidPressShoppingCart(_ urlBar: URLBarView, shoppingCart: UIButton) {
+        TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .shoppingCartButton)
+        // TODO: Update closeButtonA11yLabel for the fakespot feature
```
Comment (thatswinnie) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:115]: We should avoid having todo's in code. Could you please create a task for this?

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift [hunk 2]
```diff
@@ -109,6 +110,15 @@ extension BrowserViewController: URLBarDelegate {
...
+        let viewModel = BottomSheetViewModel(closeButtonA11yLabel: .CloseButtonTitle)
+        let bottomSheetVC = BottomSheetViewController(viewModel: viewModel,
+                                                      childViewController: FakespotBottomSheetChildViewController())
+        present(bottomSheetVC, animated: false)
```
Comment (thatswinnie) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:119]: I think we need the animation here.

Comment (PARAIPAN9) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift:119]: animation set to true does not help, it makes it fade in, but if we let it to false and making the height half of the screen, it animates from the bottom to top.

**Review by thatswinnie — APPROVED** [review]

**Comment by PARAIPAN9** [comment]
# Request for data collection review form

1) What questions will you answer with this data?
- This will help us understand if the shopping experience is used by the users

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
- By knowing if the feature is used, we will be able to evaluate its importance, and based on this, we could make improvements.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
- There are no other methods

4) Can current instrumentation answer these questions?
- No

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Records when the shopping icon from the address bar is tapped</td>
    <td>Category 2</td>
    <td>https://mozilla-hub.atlassian.net/browse/FXIOS-7054</td>
  </tr>
  <tr>
    <td>Records when the surface (bottom sheet) is dismissed</td>
    <td>Category 2</td>
    <td>https://mozilla-hub.atlassian.net/browse/FXIOS-7054</td>
  </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
 * Often the Privacy Notice for your product will link to where the documentation is expected to be.
 * Common examples for Mozilla products/services:
    * If this collection is Telemetry you can state "This collection is documented in its definitions files Histograms.json, Scalars.yaml, and/or Events.yaml and in the Probe Dictionary at https://probes.telemetry.mozilla.org."
    * If this data is collected using the Glean SDK you can state “This collection is documented in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/"
 * In some cases, documentation is included in the project’s repository.

7) How long will this data be collected?  Choose one of the following:
- Currently set to expire 2024-07-27 (~12 Months)

8) What populations will you measure?
- All channels, all locales, all countries

9) If this data collection is default on, what is the opt-out mechanism for users?
- The user can opt out

10) Please provide a general description of how you will analyze this data.
- Glean

11) Where do you intend to share the results of your analysis?
- Glean and ad partners

12) Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection? If so:
- No third party tools

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This will be reviewed by me every 6 months

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Review by razvanlitianu — APPROVED** [review]

**Pull request merged**